zipkin-storage/influxdb-http/src/main/java/zipkin/storage/influxdb/http/InfluxDBHttpStorage.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2016-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.storage.influxdb.http;
+
+import org.influxdb.InfluxDB;
+import zipkin2.Call;
+import zipkin2.DependencyLink;
+import zipkin2.Span;
+import zipkin2.storage.QueryRequest;
+import zipkin2.storage.SpanConsumer;
+import zipkin2.storage.SpanStore;
+import zipkin2.storage.StorageComponent;
+
+import java.util.List;
+
+public class InfluxDBHttpStorage extends StorageComponent implements SpanStore, SpanConsumer {
+  private InfluxDB influxDBClient;
```

Comment:
I think the client makes a network connection, so it should be lazy initialized and closed. you can look for similar code where `@Memoized` is used

Reply:
InfluxDB client supports batching

```
// Flush every 2000 Points, at least every 100ms
influxDB.enableBatch(2000, 100, TimeUnit.MILLISECONDS);
```
Do you think is it something that we can add to zipkin?

Reply:
hmm maybe, but all the more reason to lazy initialize (so that the batch processor can be closed). Can you link to the code behind this? In general it is not great to have hidden buffers, as it makes what's in flight more difficult to see.

Reply:
fwiw a collector2 module would likely duplicate this batching, for example. It matters a bit less now (provided the batching is smart enough to not write empty data)

Reply:
some other thoughts.. in tests, you'll need to disable batching or else they will fail on readback. also, probably best to hide the influxdb type (ex not accept it as a builder). this will give us flexibility later, for example if we end up with an okhttp or moshi conflict in the future, we can write native influxdb driver. for that matter, maybe best to not start with batch mode :)

Reply:
Perfect, let's keep this thinks simple and we can add more complexity if we need it